### ISSUE TITLE
Invitation link for existing user

### ISSUE THREAD

**Issue by MadameUWI** [humhub/humhub#6730]
### Is your feature request related to a problem? Please describe.
I have the following problem: I would like to send our customers an invitation link to the private space, so they can enter by themselves. The link is like https://xxx.humhub.com/user/registration/by-link?token=xxxxxxxxxxx&spaceId=19 

This link does not work for
- user who are already signed in in that moment
- users who have an account already but are not signed in
![image](https://github.com/humhub/humhub/assets/153623081/910c4738-49f7-416f-b593-c26b8912b79b)

The interface does not provide any option for these two cases. It supports registration only.

### Describe the solution you'd like
A link that
- detects if the user is signed in already
- provides a Sign-in instead-button in the registration interface

### Describe alternatives you've considered
I add existing users manually and invite new users by email manually which takes much more time.

Thank you!

**Comment by marc-farre** [comment]
> user who are already signed in in that moment

This seems to be solved (tested on HumHub 1.15.4)

> users who have an account already but are not signed in

This is a bug.
Because you can still enter your email and you will receive the "Welcome to My HumHub. Please click on the button below to proceed with your registration." email.

After clicking on the "Sign up" button, you can fill the registration form and after submitting it and error is thrown to the log:
```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062
Duplicate entry ['name@mail.com](mailto:'name@mail.com)' for key 'unique_email'
```

@luke- As we don't want hackers to have a solution to know if a user account exists for an email address, what do you think about the idea of changing the text of the email if a user already exists in the database?

E.g: "Welcome to My HumHub. Someone tried to create an account using your email. But you already have an account. Please click the button below to sign in."

**Comment by luke-** [comment]
Sounds like a good solution.

@Semir1212 What do you think? How should we design the wording of the e-mail?

**Comment by Semir1212** [comment]
@luke- @marc-farre 

"Account Creation Request

An account associated with your email address already exists. Please log in by clicking the button below.

If this request was not made by you, kindly disregard this email or reach out to your administrator for further assistance."

**Comment by luke-** [comment]
@marc-farre Would you like to create a PR for it? I would like to include this in the upcoming beta (because of translations) .

**Comment by marc-farre** [comment]
@Semir1212 thanks.
@luke- yes, I can work on it tomorrow.

However, I've discovered another issue.
If the user has already been invited by email, we have this:

![image](https://github.com/humhub/humhub/assets/23310825/2009b6b5-46f9-40c3-87cb-4c769ef0f4b9)

This is because if this rule: https://github.com/humhub/humhub/blob/2dcf8aef53656dbdd4112847937fc7ee18499fce/protected/humhub/modules/user/models/Invite.php#L77-L77

So I think I also need to check if an invitation already exists, and if yes, allow registration (either by deleting the previous invite, or by using the existing one, I'll have to see what's best).

**Comment by MadameUWI** [comment]
Hi everyone,
thank you for looking into this! 

I'm just wondering, isn't there another way than the email?

I thought more of something like this when I opened the topic:
![image](https://github.com/humhub/humhub/assets/153623081/ceab0d3f-c9da-49d2-8fb6-6288b6164b87)
After signing-in, the user gets confirmed for the space or is redirected to a page where they can hit a button to confirm the invite.

And second case, someone is already signed-in to HumHub when they click on the invitation link in the email -> they should be redirected to a page in their account and/or automatically confirmed.

From a user perspective, this would be more intuitive / simple than receiving another email. The email sounds like for a special case but in our company, we have this scenario very often. We have a community of 100 people and invite 20 of them to a new space. It is the fasted to do that through a general link via email - and this way is not well supported as described. Users are supposed to click on a button called "Register" which is a bit confusing if they know they have an account already ... (Or maybe you mean "register for a specific space" (not to HumHub in general)? But then the wording should be more clear.)

Best regards :)

**Comment by marc-farre** [comment]
@luke- PR https://github.com/humhub/humhub/pull/6916

I've discovered we already have an email template for existing registered users:
https://github.com/humhub/humhub/blob/7f40364385f961ca46ffd9e9c6d5957af9de92c5/protected/humhub/modules/user/views/mails/plaintext/UserAlreadyRegistered.php

@Semir1212 To avoid having too many email templates, do you agree if I use it for link invite instead of your suggested text?

@luke- I've added scenario constants in the `Invite` model (there was a `invite` scenario, but no related constant, and the `SOURCE_INVITE` const was used instead in rules which could create confusions).

I've also removed `$token` variable sent to email template views because they were not used (the `$registrationUrl` already contains the token).

**Comment by luke-** [comment]
> @luke- PR #6916
> 
> I've discovered we already have an email template for existing registered users: https://github.com/humhub/humhub/blob/7f40364385f961ca46ffd9e9c6d5957af9de92c5/protected/humhub/modules/user/views/mails/plaintext/UserAlreadyRegistered.php
> 
> @Semir1212 To avoid having too many email templates, do you agree if I use it for link invite instead of your suggested text?

Ahh, that's perfect, of course. Then just use this template too. 

> 
> @luke- If we stay with this original email template, there are no new messages to translate in my PR
> 
> I've added scenario constants in the `Invite` model (there was a `invite` scenario, but no related constant, and the `SOURCE_INVITE` const was used instead in rules which could create confusions).
> 
> I've also removed the `$token` variable sent to email template views because they were not used (the `$registrationUrl` already contains the token). Should I add this info on https://docs.humhub.org/docs/theme/migrate in case some custom theme uses this variable in views overwriting these email templates?

No I don't think it's neccessary. I wouldn't manage the Theme Migration Guide in such deep detail.   And also cover main changes (e.g. Layouts, Less) there.

**Comment by luke-** [comment]
@marc-farre Should I merge the PR or wait for further feedback?

**Comment by marc-farre** [comment]
> @marc-farre Should I merge the PR or wait for further feedback?

I think we already can merge it because @MadameUWI asks for a different request (adding a "Sign up" link) which can be studied later when we have more time for it.
This PR only contains bug fixes, so it's can only improve HumHub 1.16
